components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractLocalApplicationAuthenticator.java

```diff
@@ -0,0 +1,337 @@
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class AbstractLocalApplicationAuthenticator extends AbstractApplicationAuthenticator
+        implements ApplicationAuthenticator {
+
+    private static final long serialVersionUID = -4406878411547612129L;
+    private static final Log log = LogFactory.getLog(AbstractLocalApplicationAuthenticator.class);
+
+    @Override
+    public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletResponse response,
+                                           AuthenticationContext context) throws AuthenticationFailedException,
+            LogoutFailedException {
+
+        // if an authentication flow
+        if (!context.isLogoutRequest()) {
+            if (!canHandle(request)
+                    || Boolean.TRUE.equals(request.getAttribute(FrameworkConstants.REQ_ATTR_HANDLED))) {
+                return processFailedAuthenticationFlow(request, response, context);
+            } else {
+                AuthenticationDataPublisher authnDataPublisherProxy = FrameworkServiceDataHolder.getInstance()
+                        .getAuthnDataPublisherProxy();
+                Map<String, Object> unmodifiableParamMap;
+                String eventName;
+                try {
+                    if (eventFiringEnabledForAccountLocking()) {
+                        eventName = IdentityEventConstants.Event.PRE_AUTHENTICATION;
+                        fireEvent(context, eventName, false);
+                    }
+                    processAuthenticationResponse(request, response, context);
+                    if (this instanceof LocalApplicationAuthenticator && !context.getSequenceConfig()
+                            .getApplicationConfig().isSaaSApp()) {
+                        processWithNonSaaSApp(context);
+                    }
+                    request.setAttribute(FrameworkConstants.REQ_ATTR_HANDLED, true);
+                    context.setProperty(FrameworkConstants.LAST_FAILED_AUTHENTICATOR, null);
+                    if (authnDataPublisherProxy != null && authnDataPublisherProxy.isEnabled(context)) {
+                        unmodifiableParamMap = getStringObjectMap(context, context.getSubject());
+                        authnDataPublisherProxy.publishAuthenticationStepSuccess(request, context, unmodifiableParamMap);
+                        if (eventFiringEnabledForAccountLocking()) {
+                            eventName = IdentityEventConstants.Event.POST_AUTHENTICATION;
+                            fireEvent(context, eventName, true);
+                        }
+                    }
+                    return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+                } catch (AuthenticationFailedException e) {
+                    if (checkUserAccountStatus(response, context)) return AuthenticatorFlowStatus.INCOMPLETE;
```

Comment:
please refactor this if statement with proper {} as this syntax will make it hard to read the code.

```
if (checkUserAccountStatus(response, context)) {
    return AuthenticatorFlowStatus.INCOMPLETE;
}
```